compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -3104,13 +3113,13 @@ impl<'ast, 'ra, 'tcx> LateResolutionVisitor<'_, 'ast, 'ra, 'tcx> {
                 lifetime_ref.ident.span,
                 E0261,
                 "use of undeclared lifetime name `{}`",
-                lifetime_ref.ident
+                lifetime_name
             )
             .with_span_label(lifetime_ref.ident.span, "undeclared lifetime")
         };
 
         // Check if this is a typo of `'static`.
-        if edit_distance(lifetime_ref.ident.name.as_str(), "'static", 2).is_some() {
+        if edit_distance(lifetime_name.as_str(), "'static", 2).is_some() {
```

Comment:
Here (and the other call below in the diff), we should not pass the span-to-snippet to edit-distance functions since that skews / distorts the results. They will now probably consider `r#one` and `robe` to be more similar than before